### DISCUSSION TITLE
How to implement undo redo in react data grid.

### DISCUSSION THREAD

**Discussion by DVGY** [Comcast/react-data-grid#3335] [Q&A]
Hey guys is there way to implement undo redo, does react data grid support it. 
Is there a guide based on it ?

**Comment by amanmahajan7** [comment]
This is something that should be handled outside the grid. You can keep a array of rows and switch the index on undo/redo.

```
function undo() {
  index = index - 1;
}

function redo() {
  index = index + 1;
}

const currentRows = rows[index]
```

And ever change is pushed as new array
